src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -104,12 +104,25 @@ private void AddExistField(QueryFieldName fieldName, BlittableJsonReaderObject p
             IsCollectionQuery = false;
         }
 
-        public void AddWhereField(QueryFieldName fieldName, BlittableJsonReaderObject parameters, bool search = false, bool exact = false, AutoSpatialOptions spatial = null)
+        public void AddWhereField(QueryFieldName fieldName, 
+            BlittableJsonReaderObject parameters,            
+            bool search = false, 
+            bool exact = false, 
+            AutoSpatialOptions spatial = null,
+            OperatorType? operatorType = null,
+            bool isNegated = false
```

Comment:
Why do you need this? The `operatorType` should be enough, no?

Reply:
no, it isn't, see:
var query = session.Query<Bunny>().Where(u => !u.Id.In(bunnies))
We get to id equal something but we lose the fact that we are under negation